modules/identities/scripts/sign_in.ts

```diff
@@ -0,0 +1,46 @@
+import { ScriptContext } from "../module.gen.ts";
+import { IdentityDataInput, IdentityProviderInfo } from "../utils/types.ts";
+
+export interface Request {
+    info: IdentityProviderInfo;
+    uniqueData: IdentityDataInput;
+}
+
+export interface Response {
+    userToken: string;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+    const key = req.info.identityType + ":" + req.info.identityId + ":" + JSON.stringify(req.uniqueData);
+	await ctx.modules.rateLimit.throttle({
+        key,
+        period: 10,
+        requests: 10,
+        type: "user",
+    });
+
+    // Get users the provider is associated with
+    const identity = await ctx.db.userIdentities.findFirst({
+        where: {
+            identityType: req.info.identityType,
+            identityId: req.info.identityId,
+            uniqueData: { equals: req.uniqueData },
+        },
+        select: {
+            userId: true,
+        },
+    });
+
+    // If the provider info/uniqueData combo is not associated with a user,
+    // throw provider_not_found error.
+    if (!identity) {
+        throw new Error("identity_not_found");
+    }
+
+    // Generate a user token
+    const { token: { token } } = await ctx.modules.users.createToken({ userId: identity.userId });
```

Comment:
this will require a full rtt to the db to get the user id after signing in. either returh both the user token + user id + any other data you have or let the caller issue the token.

Reply:
Sounds good, will get to.